enhancements/authentication/unsupported-direct-use-of-oidc.md

```diff
@@ -0,0 +1,368 @@
+built useful toolings around the binary.
+
+## Motivation
+
+As the `oauth-server` is an authentication middle-man that mints
+OpenShift-specific access tokens, it is impossible to use tokens issued by a
+3rd party OIDC provider directly to authenticate to OpenShift clusters. Such
+a token gets forwarded directly to the `kube-apiserver` instead of the `oauth-server`,
+where the latter is at least aware of the 3rd parties that should be used as identity
+sources. But even if the token went directly to the `oauth-server`, there currently exists
+no such mechanism for the `oauth-server` to mint an OpenShift-specific access
+token based on an access or ID token from a 3rd party OIDC provider. Not to
+mention that a client should not forward its access tokens to a different entity.
+
+For the reasons above, it makes sense to allow configuring OIDC-related flags of
+the `kube-apiserver` directly in order to fully leverage capabilities of the
+community-built tools around OIDC-related authentication. However, there are some
+OpenShift specifics that one needs to bear in mind should they want to use their
+own OIDC provider.
+
+### Goals
+
+- describe expectations that an OIDC provider needs to fulfill in order for its
+  integration with OpenShift to be seamless
+- provide a guide to configure OIDC directly with a use of unsupportedConfigOverrides
+- point out which OpenShift components would be affected by the `oauth-server` disappearing
+  from the cluster and what they would need to do to integrate with a direct OIDC
+  provider instead
+
+### Non-Goals
+
+- focus on a closer integration with any specific OIDC provider
+
+## Proposal
+
+### User Stories
+
+1. I am an OIDC provider developer and I would like to try out how my product integrates
+   with OpenShift
+
+### Affected core functionality the auth team cares about
+
+Throughout the time of its existence, the OpenShift authentication stack has built
+several expectations about how OAuth clients work, which claims should be present
+in a token and several others.
+
+The following points are important to realize for any administrator that would decide
+to configure OIDC directly in their cluster:
+
+- Token revocation is an unsolved problem in the Kubernetes direct OIDC configuration. This
+  is based on vanilla Kube mostly relying on ID tokens instead of access/refresh tokens
+  for its authentication. While the revocation of the latter is [standardized](https://datatracker.ietf.org/doc/html/rfc7009),
+  revocation of ID tokens is undefined.
+  A very short lifetime (~5 min) must be imposed on ID tokens to keep such a system
+  safe.
+- Since the integrated `oauth-server` is no longer involved in the login process,
+  none of the following objects get created upon login:
+  - `OAuthAuthorizeToken`/`OAuthAccessToken`
+  - `Identity`/`User`/`Groups`
+- Neither `ServiceAccount` or `OAuthClient` and the credentials derived from them
```

Comment:
- `ServiceAccount` credentials won't be accepted by identity providers in vanilla upstream either because they are issued by kube-controller-manager.
- `OAuthClient` tokens won't be accepted by identity providers because they are issued by OpenShift `oauth-server`.

Is this differentiation correct from your point of view? If so, it may make sense to add the detail here.

Reply:
This point is about what we consider to be an OAuth2 client.

I wouldn't necessarily say that `ServiceAccounts` are not considered to be an OAuth2 client by a 3rd party IdP because they are issued by kube-controller-manager. They are not considered because these 3rd party IdPs have different means of configuring their OAuth2 clients.

That's the same for `OAuthClient` objects.